src/model_service.cpp

```diff
@@ -132,6 +132,41 @@ Status GetModelStatusImpl::getModelStatus(
     return StatusCode::OK;
 }
 
+Status GetModelsImpl::getModelsVersions(std::map<std::string, tensorflow::serving::GetModelStatusResponse>& models_versions, ModelManager& manager){
+    const std::map<std::string, std::shared_ptr<Model>>& models = manager.getModels();
+    std::map<std::string, tensorflow::serving::GetModelStatusResponse> models_versions_tmp;
+    for(auto const& model : models)
+    {
+        std::optional<int64_t> no_value_model_version;
+        tensorflow::serving::GetModelStatusRequest request;
+        GetModelStatusImpl::createGrpcRequest(model.first, no_value_model_version, &request);
+        tensorflow::serving::GetModelStatusResponse response;
+        auto status = GetModelStatusImpl::getModelStatus(&request, &response, manager);
+        models_versions_tmp.insert({model.first, response});
+    }
+    models_versions.merge(models_versions_tmp);
+
+    return StatusCode::OK;
+}
+
+Status GetModelsImpl::serializeModelsVersions2Json(const std::map<std::string, tensorflow::serving::GetModelStatusResponse>& models_versions, std::string* output){
+    std::string output_tmp;
+    for(auto const model_version : models_versions)
+    {
+        output_tmp += ("{\n\"" + model_version.first + "\" : \n");
+        std::string response_str;
+        auto status = GetModelStatusImpl::serializeResponse2Json(&model_version.second, &response_str);
+        if(status != StatusCode::OK)
+        {
+            return status;
+        }
+        output_tmp += (response_str + "}") ;
+    }
+    *output = output_tmp;
```

Comment:
Can we use reference here for the `output`? If not you need to make sure `output` is a valid pointer.